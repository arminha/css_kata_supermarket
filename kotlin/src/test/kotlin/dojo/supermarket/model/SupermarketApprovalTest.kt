package dojo.supermarket.model

import dojo.supermarket.ReceiptPrinter
import org.approvaltests.Approvals
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

class SupermarketApprovalTest {

    val cart = ShoppingCart()
    val teller = Teller(catalog)

    @Test
    fun `GIVEN apples in cart without special offer WHEN checkout THEN receipt without discount`() {
        cart.addItemQuantity(apples, 2.5)

        val receipt = teller.checksOutArticlesFrom(cart)

        Approvals.verify(ReceiptPrinter().printReceipt(receipt))
    }

    @Test
    fun `GIVEN apples in cart with 10% discount WHEN checkout THEN receipt with discount`() {
        cart.addItemQuantity(apples, 2.5)
        teller.addSpecialOffer(SpecialOfferType.TenPercentDiscount, apples, 10.0)

        val receipt = teller.checksOutArticlesFrom(cart)

        Approvals.verify(ReceiptPrinter().printReceipt(receipt))
    }

    companion object {

        private const val TOOTHBRUSH_PRICE = 0.99

        private const val APPLE_PRICE = 1.99

        val catalog = FakeCatalog()
        val toothbrush = Product("toothbrush", ProductUnit.Each)
        val apples = Product("apples", ProductUnit.Kilo)

        @JvmStatic
        @BeforeAll
        fun setup() {
            catalog.addProduct(toothbrush, TOOTHBRUSH_PRICE)
            catalog.addProduct(apples, APPLE_PRICE)
        }
    }
}
