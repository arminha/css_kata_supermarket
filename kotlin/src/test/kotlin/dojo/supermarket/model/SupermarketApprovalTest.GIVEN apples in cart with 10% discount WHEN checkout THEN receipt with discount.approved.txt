apples                              4.98
  1.99 * 2.500
10.0% off(apples)                  -0.50

Total:                              4.48